### ISSUE TITLE
HParams plugin ModuleNotFoundError v1.13.1

### ISSUE THREAD

**Issue by moritzmeister** [tensorflow/tensorboard#2122]
Installing Tensorflow/board v1.13.1 via pip, it seems like the pip package is missing the HParams plugin.

Importing it `from tensorboard.plugins import hparams` 
I get a `ModuleNotFoundError: No module named 'tensorboard.plugins.hparams'`

But the module is in the release branch of 1.13.0 here on GitHub? Am I missing something?
@wchargin can shed some light? why is the HParams plugin not included in the build?

 - TensorBoard version:
v1.13.0
 - OS Platform and version
Ubuntu 16.04
 - Python version 
v3.6.8

```
help(tensorboard.plugins)

Help on package tensorboard.plugins in tensorboard:

NAME
    tensorboard.plugins

PACKAGE CONTENTS
    audio (package)
    base_plugin
    beholder (package)
    core (package)
    custom_scalar (package)
    debugger (package)
    distribution (package)
    graph (package)
    histogram (package)
    image (package)
    interactive_inference (package)
    pr_curve (package)
    profile (package)
    projector (package)
    scalar (package)
    text (package)
```

**Comment by hgadig** [comment]
Can you try with tb-nightly and let us know if you are getting the same error?

**Comment by wchargin** [comment]
Hi @moritzmeister—you’re correct that hparams is not included in the
1.13 series. A version of the code is in the repository, but #1882 had
not merged by the time that we released TensorBoard 1.13.0, so the
plugin was not enabled or included in the Pip package. This is also why
it’s not mentioned in the release notes (`RELEASE.md`).

The hparams plugin will be in TensorBoard 1.14, and is in our nightly
builds (`pip uninstall tensorboard && pip install -U tb-nightly`).

**Comment by moritzmeister** [comment]
Thanks for the clarification @wchargin :)
When will 1.14 be released? Only with TF2.0 later this year or earlier?

**Comment by wchargin** [comment]
1.14 will be released earlier than TF 2.0, and will be compatible with
the 1.x series.

**Comment by moritzmeister** [comment]
Thanks again!
I got one last question though. I saw the [ml_demo](https://github.com/tensorflow/tensorboard/blob/d25aaf38472c9137a73611b5b2394cb26c32632b/tensorboard/plugins/hparams/hparams_ml_demo.py) you created, and I am wondering about the comment in the beginning:

>  The tag names emitted for Keras metrics changed from "acc" (in 1.x)
>  to "accuracy" (in 2.x), so this demo does not work properly in
>  TensorFlow 1.x (even with `tf.enable_eager_execution()`).

How are the metrics written by the Keras TensorBoard callback mapped to the Metrics in the HParams View?
Are the tags that you define in the experiment Proto the same as Keras uses in the summary it writes? If these tags create the connection, how can I check the tags of the metrics that Keras is writing? (So I can adjust for TF1.x)

`
metrics = [
      api_pb2.MetricInfo(
          name=api_pb2.MetricName(group="validation", tag="epoch_accuracy"),
          display_name="accuracy (val.)",
      ),
`

Thank you so much!

**Comment by wchargin** [comment]
> How are the metrics written by the Keras TensorBoard callback mapped
> to the Metrics in the HParams View? Are the tags that you define in
> the experiment Proto the same as Keras uses in the summary it writes?

Yes; metrics are identified by the tag of the associated scalar summary
(not, for instance, than the Keras-specific metric “name”).

> how can I check the tags of the metrics that Keras is writing?

The result of `model.fit` has a `history` attribute, which is a `dict`.
I can’t find any specification of what this is supposed to contain, but
empirically it appears to approximately map the tag names to their lists
of scalar values. With

```python
result = model.fit(...)
print(list(result.history))
```

the output is `['loss', 'val_accuracy', 'val_loss', 'accuracy']` in
current TF 2.0 nightly, and `['acc', 'loss', 'val_acc', 'val_loss']` in
the current TF 1.x nightly, whence you can discern what you want. But do
note that I say “approximately”, because in TF 2.0 `val_accuracy` is not
actually a tag name at all; instead, there is one `train` run and one
`validation` run; each has a scalar summary with tag `accuracy`.

Hope this helps. Sorry that I don’t have a clear-cut answer.

**Comment by wchargin** [comment]
Update: I’ve asked the Keras team; we’ll bump `tf.keras.__version__` at
the TF 2.0 release. Currently, that version is 2.2.4; as of TF 2.0, it
will likely be 2.3.

**Comment by moritzmeister** [comment]
Hey @wchargin! Thanks one more time.

I've been trying out your resent additions to the Python API. And there is one thing that I. might have misunderstood, you said in an earlier answer:

> 1.14 will (likely) be released earlier than TF 2.0, and will be
compatible with the 1.x series.

However, you are using this `tf.summary.experimental.write_raw_pb(raw_pb, step=0)`
method which is not in the TF1.x series. And also not in the TF 2.0.0a0 preview:

`module 'tensorflow._api.v2.compat.v2.summary.experimental' has no attribute 'write_raw_pb'
Traceback (most recent call last):
  File "/srv/hops/anaconda/anaconda/envs/hparams/lib/python3.6/site-packages/maggy/tensorboard.py", line 57, in write_hparams_proto
    hp.hparams_config(hparams=HPARAMS, metrics=METRICS)
  File "/srv/hops/anaconda/anaconda/envs/hparams/lib/python3.6/site-packages/tensorboard/plugins/hparams/summary_v2.py", line 131, in hparams_config
    return _write_summary("hparams_config", pb)
  File "/srv/hops/anaconda/anaconda/envs/hparams/lib/python3.6/site-packages/tensorboard/plugins/hparams/summary_v2.py", line 225, in _write_summary
    return tf.summary.experimental.write_raw_pb(raw_pb, step=0)
AttributeError: module 'tensorflow._api.v2.compat.v2.summary.experimental' has no attribute 'write_raw_pb'`

I suppose it is only in the nightly of TF2.0. Do I see this right and how is the TensorBoard 1.14 then going to work with the TF1.x series?

**Comment by nfelt** [comment]
`tf.summary.experimental.write_raw_pb` is in the tf-nightly release, and will be in TF 1.14, it just isn't out yet.  So TensorBoard 1.14 will be compatible with TF 1.14.

**Comment by wchargin** [comment]
@moritzmeister: Sorry, missed your comment earlier. In that file, we’ve
actually imported

```python
from tensorboard.compat import tf2 as tf
```

which will be equivalent to `tensorflow.compat.v2` when you have
TensorFlow 1.x installed, or `tensorflow` when you have TensorFlow 2.0
installed. On current TF 1.x nightlies, you can see:

```python
>>> import tensorflow as tf
>>> tf.__version__
'1.14.1-dev20190509'
>>> tf.compat.v2.summary.experimental.write_raw_pb
<function write_raw_pb at 0x7f6f8d05f758>
```

We do run automated tests on both 1.x and 2.0.

**Comment by wchargin** [comment]
@moritzmeister: If you’re still running into errors after upgrading to
latest `tf-nightly` and `tb-nightly`, could you please post the output
of `pip freeze | grep -e tensor -e tf- -e tb-` and the code that you’re
running to hit the error?

**Comment by ellonde** [comment]
Any updates on this, will it ever be available in TF2? 

Thanks :)

**Comment by wchargin** [comment]
@ellonde: Not sure exactly what you’re asking. TensorBoard 1.14 has now
been released, and includes the HParams plugin, as does the current
TensorFlow 2.0 beta. This thread is getting a bit hard to follow; if you
have further questions, please be explicit about what you’re asking, or
open a new thread if it’s unrelated to this one. Thanks!

**Comment by ellonde** [comment]
Arh yea, I can see that im not very specific, sorry about that :)

I was still getting `module 'tensorflow._api.v2.compat.v2.summary.experimental' has no attribute 'write_raw_pb'` as mentioned above. But i found the error, it was that I had tf2.0alpha and not beta. Its working now :+1:

**Comment by wchargin** [comment]
Great; glad to here that upgrading fixed the issue.

**Comment by MitchellProductions** [comment]
I installed tb-nightly, but I got "TypeError: 'module; object is not callable" on "hparams=hp("

`import tensorflow as tf`
`from text import symbols`
`from tensorboard.plugins import hparams as hp`

`def create_hparams(hparams_string=None, verbose=False):`
    `"""Create model hyperparameters. Parse nondefault from given string."""`

    hparams = hp(
        ################################
        # Experiment Parameters        #
        ################################
        epochs=500,
        iters_per_checkpoint=1000,
        seed=1234,
        dynamic_loss_scaling=True,
        fp16_run=False,
        distributed_run=False,
        dist_backend="nccl",
        dist_url="tcp://localhost:54321",
        cudnn_enabled=True,
        cudnn_benchmark=False,
        ignore_layers=['embedding.weight'],
